modules/tensor_mechanics/src/materials/RadialReturnStressUpdate.C

```diff
@@ -90,13 +97,24 @@ RadialReturnStressUpdateTempl<is_ad>::calculateNumberSubsteps(
 
   if (!MooseUtils::absoluteFuzzyEqual(effective_elastic_strain, 0.0))
   {
-    const Real ratio = effective_elastic_strain / _max_inelastic_increment;
 
-    if (ratio > _substep_tolerance)
-      substep_number = std::ceil(ratio / _substep_tolerance);
+    if (!_use_substep_integration_error)
+    {
+      const Real ratio = effective_elastic_strain / _max_inelastic_increment;
+
+      if (ratio > _substep_tolerance)
+        number_of_substeps = std::ceil(ratio / _substep_tolerance);
```

Comment:
We probably need to set up an upper limit for the number_of_substeps too. According to my experience with crystal plasticity, when number of sub stepping is greater than 10, it no longer helps the convergence and would slow down the calculation. I do know what is a best number to use here though. :-)

Reply:
This is a great point. @dewenyushu and I discussed this. It'd be good to know the reason why increased number of substeps doesn't help in some of these cases. I'd think that BCs and other loading may play a role in this, i.e. it may depend on the simulation setup. We can create a parameter that the user can override and throw an exception if it turns out the number is larger? I will include this change and see if there are any failures.

Reply:
A user may want to achieve a particular numerical creep integration error at the expense of more substeps... We should probably be generous with the default number of substeps. I can try 25 initially.